STYLE.md

```diff
@@ -0,0 +1,81 @@
+
+# Ops Python style guide
+
+This is the Python style guide we use for the Ops project. However, it's also the style we're converging on for other projects maintained by the Charm Tech team.
+
+We use Ruff for formatting, and run our code through the Pyright type checker. We also try to follow [PEP 8](https://peps.python.org/pep-0008/), the official Python style guide. However, PEP 8 is fairly low-level, so in addition we've come up with the following style guidelines.
+
+Of course, this is just a start! We add to this list as things come up in code review and we make a team decision.
+
+
+**Table of contents:**
+
+* [Simplicity](#simplicity)
+* [Specific decisions](#specific-decisions)
+* [Docs and docstrings](#docs-and-docstrings)
+
+
+## Simplicity
+
+### Avoid nested comprehensions
+
+"Flat is better than nested."
+
+**Don't:**
+
+```python
+units = [units for app in model.apps for unit in app.units]
+
+for current in (
+	status for status in pebble.ServiceStatus if status is not pebble.ServiceStatus.ACTIVE
+):
+	...
+```
+
+**Do:**
+
+```python
+units = []
+for app in model.apps:
+	for unit in app.units:
+		units.append(unit)
+
+active_statuses = [status for status in pebble.ServiceStatus if status != pebble.ServiceStatus.ACTIVE]
+for current in active_statuses:
+	...
+```
+
+## Specific decisions
+
+### Compare enum values by equality
+
+This is six of one, half a dozen of the other, but we've decided that saying `if color == Color.RED` is nicer than `if color is Color.RED` as `==` is more typical for integer- and string-like values, and if you do use an `IntEnum` or `StrEnum` you should use `==` anyway.
+
+Note that the [Enum HOWTO](https://docs.python.org/3/howto/enum.html#comparisons) first says they should be compared by identity, but then shows that equality is defined too, and so kind of adds to the confusion. See also this [StackOverflow question](https://stackoverflow.com/questions/25858497/should-enum-instances-be-compared-by-identity-or-equality) and [some discussion/disagreement](https://github.com/pylint-dev/pylint/issues/5356).
```

Comment:
It's nice to have this additional context here. Unfortunately, reading over it makes me think that `==` is the wrong choice (unless using `IntEnum`) and `is` is right. I have mixed feelings about using multiple `is` rather than `in`, but I guess it would be most consistent to have `is` there as well.

Looking over the do/don't examples, I also think that `status is pebble.ServiceStatus.ACTIVE` reads more nicely, although that's extremely subjective.

When this was discussed previously, were there any arguments (either way) that weren't included in the ones linked there?

Reply:
This was only really discussed briefly when Ben and I were talking about the Pebble enum/str PR. Perhaps it's worth revisiting, but maybe I only think that since I'm mildly in favour of `is`. We use `==` everywhere currently though, so it potentially creates a bunch of busywork.

Reply:
Yeah, this is definitely debatable. Let's finalise this decision during daily sync today.

Reply:
```python
import enum

ZERO = 0
ONE = 1
TWO = 2

class MyEnum(enum.Enum):
    ON = 1
    OFF = 0

class MyOtherEnum(enum.Enum):
    SMALL = 1
    LARGE = 2

assert MyEnum.ON is MyEnum.ON
assert MyEnum.ON is not MyEnum.OFF
assert MyEnum.ON is not ZERO
assert MyEnum.ON is not ONE
assert MyEnum.ON is not TWO
assert MyEnum.ON is not MyOtherEnum.SMALL
assert MyEnum.ON is not MyOtherEnum.LARGE

assert MyEnum.ON == MyEnum.ON
assert MyEnum.ON != MyEnum.OFF
assert MyEnum.ON != ZERO
assert MyEnum.ON != ONE
assert MyEnum.ON != TWO
assert MyEnum.ON != MyOtherEnum.SMALL
assert MyEnum.ON != MyOtherEnum.LARGE


class MyIntEnum(enum.IntEnum):
    OFF = 0
    ON = 1

class MyOtherIntEnum(enum.IntEnum):
    SMALL = 1
    LARGE = 2

assert MyIntEnum.ON is MyIntEnum.ON
assert MyIntEnum.ON is not MyIntEnum.OFF
assert MyEnum.ON is not ZERO
assert MyEnum.ON is not ONE
assert MyEnum.ON is not TWO
assert MyIntEnum.ON is not MyOtherIntEnum.SMALL
assert MyIntEnum.ON is not MyOtherIntEnum.LARGE

assert MyIntEnum.ON == MyIntEnum.ON
assert MyIntEnum.ON != MyIntEnum.OFF
assert MyIntEnum.ON != ZERO
assert MyIntEnum.ON == ONE  # different behaviour
assert MyIntEnum.ON != TWO
assert MyIntEnum.ON == MyOtherIntEnum.SMALL  # different behaviour
assert MyIntEnum.ON != MyOtherIntEnum.LARGE
```